src/main/java/rx/internal/operators/OperatorTakeLastOne.java

```diff
@@ -0,0 +1,162 @@
+                while (true) {
+                    //read the value of state and then try state transitions  
+                    //only if the value of state does not change in the meantime 
+                    //(in another requestMore() or onCompleted()). If the value 
+                    // has changed and we expect to do a transition still then 
+                    // we loop and try again.
+                    final int s = state.get();
+                    if (s == NOT_REQUESTED_NOT_COMPLETED) {
+                        if (state.compareAndSet(NOT_REQUESTED_NOT_COMPLETED,
+                                REQUESTED_NOT_COMPLETED)) {
+                            request(Long.MAX_VALUE);
+                            return;
+                        }
+                    } else if (s == NOT_REQUESTED_COMPLETED) {
+                        if (state.compareAndSet(NOT_REQUESTED_COMPLETED, REQUESTED_COMPLETED)) {
+                            emit();
+                            return;
+                        }
+                    } else
+                        // already requested so we exit
+                        return;
+                }
+            }
+        }
+
+        @Override
+        public void onCompleted() {
+            // CAS loop to atomically change state given that requestMore()
+            // may be acting concurrently
+            while (true) {
+                //read the value of state and then try state transitions  
+                //only if the value of state does not change in the meantime 
+                //(in another requestMore()). If the value has changed and
+                // we expect to do a transition still then we loop and try again.
+                final int s = state.get();
+                if (s == NOT_REQUESTED_NOT_COMPLETED) {
+                    if (state.compareAndSet(NOT_REQUESTED_NOT_COMPLETED, NOT_REQUESTED_COMPLETED)) {
+                        return;
+                    }
+                } else if (s == REQUESTED_NOT_COMPLETED) {
+                    if (state.compareAndSet(REQUESTED_NOT_COMPLETED, REQUESTED_COMPLETED)) {
+                        emit();
+                        return;
+                    }
+                } else
+                    // already completed so we exit
+                    return;
+            }
+        }
+
+        /**
+         * If not unsubscribed then emits last value and completed to the child subscriber.
+         */
+        private void emit() {
+            if (isUnsubscribed()) {
+                // release for gc
+                last = null;
+                return;
+            }
+            // synchronize to ensure that last is safely published
+            synchronized (this) {
+                if (last != ABSENT) {
+                    T t = last;
+                    // release for gc 
+                    last = null;
+                    child.onNext(t);
```

Comment:
Since you are calling ```child.onNext``` from ```onCompleted```, you need to bounce back any thrown exceptions because throwing ```onCompleted``` is can't be routed back to ```onError``` and child will not receive the exception.